core/src/main/java/bio/terra/pearl/core/dao/datarepo/DatasetDao.java

```diff
@@ -29,6 +29,10 @@ public Optional<Dataset> findByDataRepoId(UUID dataRepoId) {
         return findByProperty("dataset_id", dataRepoId);
     }
 
+    public Optional<Dataset> findByDatasetName(String datasetName) {
```

Comment:
Even though dataset names are unique, I would prefer to look this up by the Juniper-generated UUID of the dataset. In order to do so, I need to add a FK to the `data_repo_job` table. I'll do this in an upcoming PR.